test/test_deadseeker.py

```diff
@@ -0,0 +1,125 @@
+from deadseeker import DeadSeeker, SeekerConfig, LinkAcceptorBuilder
+import socket
+from threading import Thread
+import os
+from typing import ClassVar, List
+from logging import DEBUG
+import logging
+
+DIRECTORY = os.path.join(os.path.dirname(__file__), "mock_server")
+
+
+class MockServerRequestHandler(SimpleHTTPRequestHandler):
+    def translate_path(self, path):
+        return DIRECTORY + path
+
+    def do_HEAD(self):
+        if not self.check_error():
+            super().do_HEAD()
+
+    def do_GET(self):
+        if not self.check_error():
+            super().do_GET()
+
+    def check_error(self):
+        if self.path.endswith('/page4.html'):
+            self.send_response(HTTPStatus.INTERNAL_SERVER_ERROR)
+            self.end_headers()
+            return True
+        return False
+
+
+def get_free_port():
+    s = socket.socket(socket.AF_INET, type=socket.SOCK_STREAM)
+    s.bind(('localhost', 0))
+    address, port = s.getsockname()
+    s.close()
+    return port
+
+
+class TestDeadSeeker(unittest.TestCase):
+    url: ClassVar[str]
+    mock_server: ClassVar[HTTPServer]
+    mock_server_thread: ClassVar[Thread]
+
+    @classmethod
+    def setUpClass(cls):
+        # Configure mock server.
+        port = get_free_port()
+        cls.mock_server = HTTPServer(
+            ('localhost', port), MockServerRequestHandler)
+
+        # Start running mock server in a separate thread.
+        # Daemon threads automatically shut down when the main process exits.
+        cls.mock_server_thread = Thread(target=cls.mock_server.serve_forever)
+        cls.mock_server_thread.setDaemon(True)
+        cls.mock_server_thread.start()
+        cls.url = f'http://localhost:{port}'
+
+    def setUp(self):
+        self.config = SeekerConfig()
+        self.logger = logging.getLogger('deadseeker.deadseeker')
+        self.builder = LinkAcceptorBuilder()
+
+    def test_numberFailedIs2(self):
+        self.builder.addExcludePrefix('https://www.google.com')
+        response = self._seek_with_logging()
+        actualfailed = len(response.failures)
+        self.assertEqual(2, actualfailed)
+
+    def test_numberFailedIs0WithMoreExcludes(self):
+        self.builder.addExcludePrefix(
+            'https://www.google.com',
+            '/page3.html',
+            '/page4.html')
+        response = self._seek_with_logging()
+        actualfailed = len(response.failures)
+        self.assertEqual(0, actualfailed)
+
+    def test_messagesLogged(self):
+        self.builder.addExcludePrefix('https://www.google.com')
+        with patch.object(self.logger, 'error') as error_mock,\
+                patch.object(self.logger, 'info') as info_mock:
+            self._seek_with_logging()
+            expected_errors = [
+                f'::error ::ClientResponseError: 500 - {self.url}/page4.html',
+                f'::error ::ClientResponseError: 404 - {self.url}/page3.html'
+            ]
+            actual_errors: List[str] = []
```

Comment:
Maybe we should use sets instead of lists here so the order in which we specify the expected errors (or expected info) doesn't matter.

Reply:
True that sets could be used in case we see results come back in a different order; however, from my local testing the result order is surprisingly predictive.  I thought that as long as we didn't see results naturally start coming back in different orders, then it might not hurt to have the added benefit of checking the order as well.  But on the other hand, I see what you mean -- if we actually don't care about order then sets would be the way to go.  The only problem I see with sets is that the test could still pass if the code started crawling the same pages over again -- so sets introduce the risk of a false positive test case

Reply:
Ahh, I think I spoke it into existence!  The test runs fine on my local machine, but the order is different while running it in the github workflow.  This is going to be tricky with the info messages that contain the unpredictable elapsed time numbers, but maybe we can come up with something...

Reply:
It is not pretty, but I came up with a way to check the prefixes without requiring the order to be exact, looks like the workflow finished this time